src/java/org/apache/cassandra/cql3/QueryEvents.java

```diff
@@ -250,10 +242,19 @@ public void notifyPrepareFailure(@Nullable CQLStatement statement, String query,
         }
     }
 
-    private String possiblyObfuscateQuery(CQLStatement statement, String query)
+    private String maybeObfuscatePassword(CQLStatement statement, String query)
     {
         // Statement might be null as side-effect of failed parsing, originates from QueryMessage#execute
-        return null == statement || statement instanceof AuthenticationStatement ? passwordObfuscator.obfuscate(query) : query;
+        String res;
+
+        if (statement == null)
+            res = PasswordObfuscator.obfuscate(query);
+        else if (statement instanceof AuthenticationStatement)
+            res = ((AuthenticationStatement) statement).obfuscatePassword(query);
+        else
+            res = query;
+
+        return res;
```

Comment:
This code can be simplified to:
```
if (statement == null)
    return PasswordObfuscator.obfuscate(query);

if (statement instanceof AuthenticationStatement)
     return ((AuthenticationStatement) statement).obfuscatePassword(query);

return query;
```
It is much easier to read.

Reply:
Funny. I'd swore I replied already to this in the 4.0 PR but I can't find my answer anywhere now.

This is intentional for performance. On branching you want to avoid returning results as this breaks CPU branching optimizations/prediction. You want the construct I used (as per the perf training we did)

Reply:
I do not believe that the impact of branch misprediction will be high in this case. This part of code is also not on the critical path so I would clearly prefer clarity. :-)

Reply:
I read my option clearer. I think it comes down to personal taste. Changed and pushed. Waiting for CI now.